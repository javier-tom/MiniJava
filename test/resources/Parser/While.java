class Main {
    public static void main(String[] args) {
        System.out.println(new Loops().loops(5));
    }
}

class Loops {
    public int loops(int max) {
        int i;
        i = 0;
        while (i < max) {
            System.out.println(i);
            i = i + 1;
        }
        i = 0;

        return 0;
    }
}