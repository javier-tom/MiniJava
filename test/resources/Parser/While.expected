Program
  MainClass Main (line 1)
    Print (line 3)
      Call loops (line 3)
        on class:
          NewObject Loops (line 3)
        parameters:
          5 (line 3)

  Class Loops (line 7)
    fields:
    MethodDecl loops (line 8)
      returns int
      params:
        int max
      variables:
        VarDecl int i (line 9)
      Assign i (line 10)
        new value:
          0 (line 10)
      While (line 11)
        condition:
          LessThan (line 11)
            i (line 11)
            max (line 11)
        Print (line 12)
          i (line 12)
        Assign i (line 13)
          new value:
            Plus (line 13)
              i (line 13)
              1 (line 13)
      Assign i (line 15)
        new value:
          0 (line 15)
      Return
        0 (line 17)
